support-workers/src/test/scala/com/gu/support/workers/lambdas/UpdateSupporterProductDataSpec.scala

```diff
@@ -21,33 +21,36 @@ import scala.io.Source
 class UpdateSupporterProductDataSpec extends AnyFlatSpec {
 
   "UpdateSupporterProductData" should "not insert an item into Dynamo for a digisub gift purchase" in {
-    val state = decode[SendThankYouEmailState](digitalSusbcriptionGiftPurchaseState)
-    state.isRight shouldBe true
-    val supporterRatePlanItem = UpdateSupporterProductData
-      .getSupporterRatePlanItemFromState(state.toOption.get, serviceWithFixtures)
-    inside(supporterRatePlanItem) { case Right(value) =>
-      value shouldBe None
+    val maybeState = decode[SendThankYouEmailState](digitalSusbcriptionGiftPurchaseState)
+    inside(maybeState) { case Right(state) =>
+      val supporterRatePlanItem =
+        UpdateSupporterProductData.getSupporterRatePlanItemFromState(state, serviceWithFixtures)
+      inside(supporterRatePlanItem) { case Right(value) =>
+        value shouldBe None
+      }
     }
   }
 
   "UpdateSupporterProductData" should "return a valid SupporterRatePlanItem for a digisub gift redemption" in {
-    val state = decode[SendThankYouEmailState](digitalSubscriptionGiftRedemptionState)
-    state.isRight shouldBe true
-    val supporterRatePlanItem = UpdateSupporterProductData
-      .getSupporterRatePlanItemFromState(state.toOption.get, serviceWithFixtures)
-    inside(supporterRatePlanItem) { case Right(item) =>
-      item.value.identityId shouldBe "102803446"
+    val maybeState = decode[SendThankYouEmailState](digitalSubscriptionGiftRedemptionState)
+    inside(maybeState) { case Right(state) =>
```

Comment:
I know some of this is copy / paste, but I find ScalaTest's `Values` `traits` useful in creating easy to read test avoiding nested assertions. e.g. [`EitherValues`](https://www.scalatest.org/user_guide/using_EitherValues)

Reply:
it's not even copy and paste, the main change is indentation, I haven't actually added anything material.  I'm always up for learning more scalatest stuff though so thanks for the pointer!